### DISCUSSION TITLE
Already planed? : about preparing schema and migration

### DISCUSSION THREAD

**Discussion by roccho-dev** [electric-sql/pglite#210] [Ideas]
From reading the documentation, it is assumed that the schema is defined after the application is launched.

If I don't use 'electricSQL', is there an option planned for this Repository or project to provide the files after the schema definition or migration in advance?
(I assume this is not a solution of the nature I am asking, but I am sure there will be developers who will have the same question) (apologies if this has already been answered)

**Comment by AntonOfTheWoods** [comment]
I'm not sure I understand what you are asking but if you are asking, can you predefine the schema and data and load pglite directory onto an existing database, then yes, at least according to the documentation!

https://pglite.dev/docs/api#dumpdatadir

With this you can create a postgres datadir and then just supply a tar.gz of that to your new instance.

**Comment by samwillis** [comment]
Hey @PorcoRosso85 

Good question, we don't have any standard schema migration tooling built in (yet). The easers is to just run you migrations immediately after starting PGLite:

```ts
const pg = await PGlite.create()

pg.exec(`
  CREATE TABLE IF NOT EXIST my_table (
    -- ... schema
  )  
  
  CREATE TABLE IF NOT EXIST my_other_table (
    -- ... schema
  )  
`)

// Now you can use the db
```

There are many tools that do schema migration in different ways, and with ORMd such a Drizzle supporting PGlite, using their migration tooling is likely to be preferred in most cases.

However, I am considering a basic migration plugin modelled on [@databases/pg-migrations](https://www.atdatabases.org/docs/pg-migrations), but this is fairly low down the priority list. It is like better served by the community.

**Reply by AntonOfTheWoods** [reply]
@samwillis , can you point to the `drizzle` docs that show how to do migrations with `pglite` (or `wa-sqlite`)? From memory it wasn't possible (https://github.com/drizzle-team/drizzle-orm/issues/1009) and I can't find how to do it in their current docs.

**Comment by roccho-dev** [comment]
Thanks for your reply.


@AntonOfTheWoods 
I just want to prepare schema of 'pglite' because of ux.
Creating schema require some cost, I felt on demo.

@samwillis @AntonOfTheWoods 
for example, is it not  possible yet? as follows:
- to skip creating the schema on the client side by loading the data including the schema from the already hosted postgres server
- Alternatively, we can pre-define and persist the schema in a wasm file, host it, and load it by client-side.
